### PR TITLE
[FLINK-33738][Scheduler] Make exponential-delay restart-strategy the default restart strategy

### PR DESCRIPTION
**Pull request by 1996fanrui**
[FLINK-33738][Scheduler] Make exponential-delay restart-strategy the default restart strategy

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java [hunk 1]
```diff
@@ -89,11 +89,10 @@ public class RestartStrategyOptions {
                                                             "here")))
                                     .text(
                                             "If checkpointing is disabled, the default value is %s. "
-                                                    + "If checkpointing is enabled, the default value is %s with %s restart attempts and '%s' delay.",
+                                                    + "If checkpointing is enabled, the default value is %s, and the values of all %s related options have not changed.",
```
Comment (1996fanrui) [inline flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java:92]: Hi @zhuzhurk , do you have any good suggestion about the doc of default restart strategy?

**Review by 1996fanrui — COMMENTED** [review]
Hi @zhuzhurk @RocMarshal @qinf , would you mind helping take a look this PR in your free time? Thanks~

**Comment by qinf** [comment]
Thank you @1996fanrui for your contribution, LGTM.

**Review by qinf — APPROVED** [review]

File: flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java (hunk 1 above)
Comment (zhuzhurk) [reply flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java:92]: maybe "and the values of all %s related options have not changed." -> "and the default values of %s related config options will be used."

Comment (1996fanrui) [reply flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java]: Thanks for the suggestion! It's better than before, I updated.

**Review by zhuzhurk — APPROVED** [review]
LGTM :)

**Review by RocMarshal — APPROVED** [review]
Thanks for the contribution~
LGTM +1.  :)

**Comment by 1996fanrui** [comment]
Thanks to everyone for the review! The CI is green for now, merging~

**Pull request merged**